Frameworks/UIKit/UIScrollView.mm

```diff
@@ -169,87 +162,29 @@ - (id)_dManipGestureCallback:(_UIDMPanGestureRecognizer*)gesture {
 
 - (void)_initUIScrollView {
     // Store a strongly-typed backing scrollviewer
-    _scrollViewer = rt_dynamic_cast<WXCScrollViewer>([self xamlElement]);
-    if (!_scrollViewer) {
+    WXCGrid* scrollViewGrid = rt_dynamic_cast<WXCGrid>([self xamlElement]);
+    WXCScrollViewer* scrollViewer = nil;
+
+    if (scrollViewGrid) {
+        scrollViewer = XamlControls::GetScrollViewer(scrollViewGrid);
+    } else {
+        scrollViewer = rt_dynamic_cast<WXCScrollViewer>([self xamlElement]);
+    }
+
+    if (!scrollViewer) {
+        // Definitely didn't receive any supported backing xaml elements
         FAIL_FAST();
     }
 
+    _scrollViewer = scrollViewer;
+
     self->_dManipGesture = [[_UIDMPanGestureRecognizer alloc] initWithTarget:self action:@selector(_dManipGestureCallback:)];
     [self addGestureRecognizer:self->_dManipGesture];
 
     [self setMultipleTouchEnabled:TRUE];
 
-    // create insets, assign names to make it easy to debug in visual tree
-    _topInset = [WUXSRectangle make];
```

Comment:
since these are completely removed in the new implementation, so definitely improvement. 

---
In reply to: [106568739](https://github.com/Microsoft/WinObjC/pull/2257#discussion_r106568739) [](ancestors = 106568739)